docs/resource-guide.md

```diff
@@ -60,7 +60,10 @@ it is not specified, but others cannot, and some might poorly infer if the
 struct name is ambiguous.
 
 If you'd like your resource to be accessible by the `YAML` graph API (GAPI),
-then you'll need to include the appropriate YAML fields as shown below.
+then you'll need to include the appropriate YAML fields as shown below. This is
+used by the `Puppet` compiler as well, so make sure you include these struct
```

Comment:
For puppet-guide.md I still have another branch cooking, that updates a lot of other things also. So I'm not touching that one in this PR; it will come separately.